src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -264,39 +272,59 @@ public BitbucketApi buildBitbucketClient() {
         return getBitbucketConnector().create(repoOwner, repository, getScanCredentials());
     }
 
-    @Override
-    protected void retrieve(SCMHeadObserver observer, final TaskListener listener) throws IOException,
-            InterruptedException {
+    public BitbucketApi buildBitbucketClient(PullRequestSCMHead head) {
+        return getBitbucketConnector().create(head.getRepoOwner(), head.getRepository(), getScanCredentials());
+    }
 
+    @Override
+    protected void retrieve(@CheckForNull SCMSourceCriteria criteria, @NonNull SCMHeadObserver observer,
+                            @CheckForNull SCMHeadEvent<?> event, @NonNull TaskListener listener)
+            throws IOException, InterruptedException {
         StandardUsernamePasswordCredentials scanCredentials = getScanCredentials();
         if (scanCredentials == null) {
-            listener.getLogger().format("Connecting to %s with no credentials, anonymous access%n", bitbucketServerUrl == null ? "https://bitbucket.org" : bitbucketServerUrl);
+            listener.getLogger().format("Connecting to %s with no credentials, anonymous access%n", bitbucketServerUrl());
         } else {
-            listener.getLogger().format("Connecting to %s using %s%n", bitbucketServerUrl == null ? "https://bitbucket.org" : bitbucketServerUrl, CredentialsNameProvider.name(scanCredentials));
+            listener.getLogger().format("Connecting to %s using %s%n", bitbucketServerUrl(), CredentialsNameProvider.name(scanCredentials));
         }
 
         // Search branches
-        retrieveBranches(observer, listener);
+        retrieveBranches(criteria, observer, listener);
         // Search pull requests
-        retrievePullRequests(observer, listener);
+        retrievePullRequests(criteria, observer, listener);
     }
 
-    private void retrievePullRequests(SCMHeadObserver observer, final TaskListener listener) throws IOException {
+    private void retrievePullRequests(SCMSourceCriteria criteria, SCMHeadObserver observer, final TaskListener listener)
+            throws IOException, InterruptedException {
         String fullName = repoOwner + "/" + repository;
         listener.getLogger().println("Looking up " + fullName + " for pull requests");
 
         final BitbucketApi bitbucket = getBitbucketConnector().create(repoOwner, repository, getScanCredentials());
         if (bitbucket.isPrivate()) {
             List<? extends BitbucketPullRequest> pulls = bitbucket.getPullRequests();
             for (final BitbucketPullRequest pull : pulls) {
+                checkInterrupt();
```

Comment:
Why is this explicit check needed? Is this running in a different thread from the main scan thread?